examples/Cassie/cassie_state_estimator.cc

```diff
@@ -232,60 +263,60 @@ CassieRbtStateEstimator::CassieRbtStateEstimator(
 ///   The connection point of the rod and the spring does not lie on the line
 ///   where the spring lies. Instead, there is a small offset.
 ///   We account for this offset by `spring_rest_offset`.
-void CassieRbtStateEstimator::solveFourbarLinkage(const VectorXd& q,
-    double* left_heel_spring, double* right_heel_spring) const {
+void CassieStateEstimator::solveFourbarLinkage(
+    const VectorXd& q, double* left_heel_spring,
+    double* right_heel_spring) const {
   // Get the spring length
   double spring_length = rod_on_heel_spring_.norm();
   // Spring rest angle offset
-  double spring_rest_offset = atan(
-                              rod_on_heel_spring_(1) / rod_on_heel_spring_(0));
-
+  double spring_rest_offset =
+      atan(rod_on_heel_spring_(1) / rod_on_heel_spring_(0));
   std::vector<Vector3d> rod_on_thigh{rod_on_thigh_left_, rod_on_thigh_right_};
-  std::vector<int> thigh_ind{left_thigh_idx_, right_thigh_idx_};
-  std::vector<int> heel_spring_ind{left_heel_spring_idx_,
-                                   right_heel_spring_idx_};
 
-  KinematicsCache<double> cache = tree_.doKinematics(q);
+  plant_.SetPositions(context_.get(), q);
 
   for (int i = 0; i < 2; i++) {
     // Get thigh pose and heel spring pose
-    const Isometry3d thigh_pose = tree_.CalcBodyPoseInWorldFrame(
-                                    cache, tree_.get_body(thigh_ind[i]));
-    const Vector3d thigh_pos = thigh_pose.translation();
-    const MatrixXd thigh_rot_mat = thigh_pose.linear();
+    auto thigh_pose = plant_.EvalBodyPoseInWorld(*context_, *(thighs_[i]));
+    const Vector3d& thigh_pos = thigh_pose.translation();
+    const auto& thigh_rot_mat = thigh_pose.rotation();
 
-    const Isometry3d heel_spring_pose = tree_.CalcBodyPoseInWorldFrame(cache,
-                                        tree_.get_body(heel_spring_ind[i]));
-    const Vector3d r_heel_spring_base = heel_spring_pose.translation();
-    const MatrixXd heel_spring_rot_mat = heel_spring_pose.linear();
+    auto heel_spring_pose =
+        plant_.EvalBodyPoseInWorld(*context_, *heel_springs_[i]);
+    const Vector3d& r_heel_spring_base = heel_spring_pose.translation();
+    const auto& heel_spring_rot_mat = heel_spring_pose.rotation();
 
     // Get r_heel_spring_base_to_thigh_ball_joint
     Vector3d r_ball_joint = thigh_pos + thigh_rot_mat * rod_on_thigh[i];
     Vector3d r_heel_spring_base_to_thigh_ball_joint =
         r_ball_joint - r_heel_spring_base;
     Vector3d r_thigh_ball_joint_wrt_heel_spring_base =
-        heel_spring_rot_mat.transpose() * r_heel_spring_base_to_thigh_ball_joint;
+        heel_spring_rot_mat.transpose() *
+        r_heel_spring_base_to_thigh_ball_joint;
 
     // Get the projected rod length in the xy plane of heel spring base
-    double projected_rod_length = sqrt(pow(rod_length_, 2) -
-        pow(r_thigh_ball_joint_wrt_heel_spring_base(2), 2));
+    double projected_rod_length =
```

Comment:
let's not use `pow()` to compute squares, I think `rod_length_ * rod_length_` would be easier to read.